### DISCUSSION TITLE
SSL Certificate Error While Downloading Packages from GitHub with cURL Body

### DISCUSSION THREAD

**Discussion by everforyou** [community/community#135956] [Programming Help]
### Body

I'm encountering an SSL certificate error while trying to download Laravel packages using Composer. Here’s the error message I receive:
```
SSL: no alternative certificate subject name matches target host name 'codeload.github.com'
Failed to download laravel/breeze from dist: curl error 60 while downloading https://codeload.github.com/laravel/breeze/legacy.zip
```

- I'm using Composer to manage my Laravel project dependencies.
- The issue arises during the syncing process of various Laravel packages, such as laravel/breeze, league/flysystem, and nette/schema.
- The error occurs consistently for multiple packages.

Steps Taken :
- Verified that the URLs in my configuration are correct.
- Updated cURL and OpenSSL to their latest versions.
- Attempted to disable SSL verification using the -k option, but this is not a recommended solution for production.

How can I fix this problem : 
- Has anyone experienced similar issues?
- What steps can I take to resolve this SSL certificate validation problem?
- Are there any best practices for handling SSL errors in Composer or cURL?

Any guidance or advice would be greatly appreciated!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Youssef-Meziani** [comment]
To resolve the SSL certificate error while downloading packages with Composer, try the following steps:

1. **Update CA Certificates:** Ensure your system’s CA certificates are up to date. On Ubuntu, you can update them with:
`sudo apt-get update && sudo apt-get install --reinstall ca-certificates`

2. **Use the --prefer-dist Option:** Force Composer to use GitHub’s API instead of cURL by running:
`composer update --prefer-dist`

3. **Check Time and Date:** Ensure your system's time and date are correct, as SSL certificates rely on accurate time.

4. **Composer CA Bundle:** Use Composer's CA bundle by setting the environment variable:
`export COMPOSER_CAFILE=/path/to/cacert.pem`

5. **GitHub Hosts:** Make sure the codeload.github.com URL isn't blocked or redirected in your /etc/hosts file.

If the problem persists, you might need to configure your server’s SSL settings or consider switching to a different network to rule out any connection issues.

**Reply by everforyou** [reply]
Thanks, I will try once.

**Comment by Alirezaseyed** [comment]
To resolve the SSL certificate error you’re encountering while downloading Laravel packages with Composer, you can follow these steps:

1. Update cURL and OpenSSL

	•	Update cURL and OpenSSL: Ensure that your cURL and OpenSSL installations are up to date. Older versions might have issues with SSL validation.
	•	On Windows, you can download the latest versions from their official websites.
	•	On Linux, you can update them using the following commands:

`sudo apt-get update
sudo apt-get upgrade curl openssl`


2. Update Composer

	•	Update Composer: A newer version of Composer might address SSL-related issues. Update Composer to the latest version:


`composer self-update`

3. Check CA Certificates

	•	Verify CA Certificates: Ensure that the CA certificates on your system are up to date. On Linux systems, these files are usually located in /etc/ssl/certs and may need to be updated.

4. Use the --insecure Option as a Temporary Fix

	•	Disable SSL Verification Temporarily: You can temporarily bypass SSL verification by using the --insecure or -k option with Composer. Note that this is not recommended for production environments as it reduces security:


`composer install --no-plugins --no-scripts --ignore-platform-reqs`


5. Check DNS and Proxy Settings

	•	Verify DNS and Proxy Settings: Ensure that your DNS and proxy settings are configured correctly. Issues with DNS or proxy settings can cause SSL problems.

6. Check php.ini Settings

	•	Inspect php.ini Settings: If Composer relies on PHP, ensure that SSL-related settings in the php.ini file are correctly configured. For example, the CA file path should be correctly set:


`openssl.cafile="/path/to/cacert.pem"`

7. Contact Support

	•	GitHub Support: If none of the above solutions work, there might be an issue with GitHub’s servers. In this case, you can contact GitHub support for assistance.

By following these steps, you should be able to resolve the SSL certificate error and successfully download packages.
